File: abc/abc221/c.py
import itertools

n = list(map(int, input()))


def to_number(lst):
    return int("".join(map(str, lst)))


ans = 0
for perm in itertools.permutations(n):
    for i in range(1, len(n)):
        left = to_number(list(perm)[:i])
        right = to_number(list(perm)[i:])
        ans = max(ans, left * right)

print(ans)

File: abc/abc221/c.rs
use proconio::input;
use itertools::Itertools;

fn main() {
    input! {
        n: String,
    }

    let permutations = n.chars().permutations(n.len());

    let mut ans = 0;
    for perm in permutations {
        let str = perm.iter().collect::<String>();
        for i in 1..n.len() {
            let (l, r) = str.split_at(i);
            let l: usize = l.parse().unwrap();
            let r: usize = r.parse().unwrap();
            ans = ans.max(l * r);
        }
    }

    println!("{}", ans);
}
